VideoWeb/VideoWeb/ClientApp/src/app/services/conference/conference.service.ts

```diff
@@ -0,0 +1,163 @@
+                map(() => activatedRoute.snapshot),
+                map(route => {
+                    while (route && !route.paramMap?.has('conferenceId')) {
+                        route = route?.firstChild;
+                    }
+
+                    return route?.paramMap;
+                })
+            )
+            .subscribe(paramMap => {
+                this.onRouteParamsChanged(paramMap);
+            });
+    }
+
+    private _currentConference: ConferenceResponse;
+    get currentConference(): ConferenceResponse {
+        return this._currentConference;
+    }
+
+    private currentConferenceSubject = new ReplaySubject<ConferenceResponse>(1);
+    get currentConference$(): Observable<ConferenceResponse> {
+        return this.currentConferenceSubject.asObservable();
+    }
+
+    private onCurrentConferenceStatusChangedSubject = new ReplaySubject<ConferenceStatusChanged>(1);
+    get onCurrentConferenceStatusChanged$() {
+        return this.onCurrentConferenceStatusChangedSubject.asObservable();
+    }
+
+    private _currentConferenceId: string;
+    get currentConferenceId(): string {
+        return this._currentConferenceId;
+    }
+
+    getConferenceById(conferenceId: string | Guid): Observable<ConferenceResponse> {
+        this.logger.info(`${this.loggerPrefix} getting conference by ID: ${conferenceId}`);
+
+        return this.apiClient.getConferenceById(conferenceId.toString());
+    }
+
+    getParticipantsForConference(conferenceId: Guid | string): Observable<ParticipantModel[]> {
+        this.logger.info(`${this.loggerPrefix} getting participants for conference.`);
+
+        return this.apiClient
+            .getParticipantsByConferenceId(conferenceId.toString())
+            .pipe(
+                map(participants =>
+                    participants.map(participantResponse => ParticipantModel.fromParticipantForUserResponse(participantResponse))
+                )
+            );
+    }
+
+    getEndpointsForConference(conferenceId: Guid | string): Observable<ParticipantModel[]> {
+        this.logger.info(`${this.loggerPrefix} getting endpoints for conference.`);
+
+        return this.apiClient
+            .getVideoEndpointsForConference(conferenceId.toString())
+            .pipe(
+                map(participants =>
+                    participants.map(videoEndpointResponse => ParticipantModel.fromVideoEndpointResponse(videoEndpointResponse))
+                )
+            );
+    }
+
+    getLoggedInParticipantForConference(conferenceId: Guid | string): Observable<ParticipantModel> {
+        return this.getParticipantsForConference(conferenceId).pipe(
+            mergeMap(participants =>
+                this.apiClient
+                    .getCurrentParticipant(conferenceId.toString())
+                    .pipe(map(response => participants.find(participant => participant.id === response.participant_id)))
+            ),
+            take(1)
+        );
+    }
```

Comment:
Might need you to explain this to me.

Reply:
So I want the actual participant model rather than the response from "getCurrentParticipant". So I need the participant responses from "getParticipantsForConference" then I can map it  to the correct representation. The merge map me return the observable for the second query ("getCurrentParticipant") but only once the first observable has completed. I hope that somewhat explains it :D